base/broadcast.jl

```diff
@@ -10,6 +10,11 @@ export broadcast_getindex, broadcast_setindex!
 
 ## Broadcasting utilities ##
 
+# fallback routines for broadcasting with no arguments or with scalars
+# to just produce a scalar result:
+broadcast(f) = f()
+broadcast(f, x::Number...) = f(x...)
```

Comment:
Oh, right, it's turning `foo.(1)` into `broadcast(() -> foo(1))` because the literal gets inlined into the function by `compress-fuse`.